integration_test/cases/type.exs

```diff
@@ -91,13 +104,15 @@ defmodule Ecto.Integration.TypeTest do
     assert [^datetime] = TestRepo.all(query)
   end
 
+  @tag :coalesce_type
   test "coalesce type when default" do
```

Comment:
Both :decimal_type_cast and :db_engine_can_maintain_precision seem to be related to decimal precision, so what if we have a @tag :decimal_precision for all of them?

Do the coalesce tests pass when they don't use decimal?

Reply:
coalesce_type is issue where we map string to varbinary. This is part that I could not fix with [migration override](https://github.com/mjaric/ecto_sql/blob/tds-support/integration_test/mssql/migration.exs) 

1. If I don not override migration this MSSQL will complain about implicit conversion nvarchar to varbinary in about 20 test cases.  binary <<0, 2>> parameter tagged as type :string  (thar is nvarchar) and column is varbinary
2. if I override migration and set column to nvarchar (:string) in migration, then this 20 test passes except these 2 with tag :coalesce_type. And here problem is that decoder needs to conver ucs2 to utf-8 (fixed length to variable length) and that is how we get <<2>> in return, padding zero is removed during decoding.

Maybe if I rename tag to :string_encoding?

Reply:
Good point. :string_encoding or :varchar_encoding works. Your call!

Reply:
Also, remember to rename the one below too. :)